src/Components/Facility/ConsultationForm.tsx

```diff
@@ -545,9 +600,19 @@ export const ConsultationForm = (props: any) => {
   //     dispatch({ type: "set_form", form });
   //   }
 
-  const handleDateChange = (date: MaterialUiPickersDate, key: string) => {
+  const handleDateChange = (
+    date: MaterialUiPickersDate,
+    key: string,
+    nested?: string[]
+  ) => {
+    let newValue: any = { [key]: date };
+    if (nested?.length) {
+      nested?.forEach((i: string) => {
+        newValue = { [i]: newValue };
+      });
```

Comment:
just need some clarification here,
```js
// if we have something like this
handleDateChange(new Date(), 'now', ['procedure', 'extra']);

// then new value becomes
// initial
newValue = { now: new Date() }
// procedure
newValue = {procedure: { now: new Date() }}
// extra
newValue = {extra: {procedure: { now: new Date() }}}
```

So, if my understanding is correct, why are we nesting them?

Reply:
Hey @khavinshankar, it was just an approach to handle the date changes for the nested elements as well... when they are not at the root level of the form object but inside another object.
Nesting just checks if that is the case and traces backwards and creates an updated object from that level to the root.